apps/analysis_framework/serializers.py

```diff
@@ -708,3 +708,25 @@ def create(self, validated_data):
         validated_data['framework'] = self.framework
         validated_data['added_by'] = self.context['request'].user
         return super().create(validated_data)
+
+
+class AnalysisFrameworkCloneSerializer(serializers.Serializer):
+    title = serializers.CharField(required=True)
+    description = serializers.CharField(required=False)
+    project = serializers.PrimaryKeyRelatedField(queryset=Project.objects.all(), required=False)
+
+    def validate_project(self, project):
+        if not project.can_modify(self.context['request'].user):
+            raise serializers.ValidationError('User does not have permission to modify the project')
+        return project
+
+    def create(self, validated_data):
+        af = self.context['request'].active_af
```

Comment:
We need this check as well
https://github.com/the-deep/server/blob/e951e7892d1618a28d02d92b544f5e788980d7d2/apps/analysis_framework/views.py#L100-L101